recipes/cad1/task2/baseline/evaluate.py

```diff
@@ -0,0 +1,327 @@
+        )
+
+    audio_manager.add_audios_to_save("ha_processed_signal", processed_signal)
+
+    # 7. Normalise reference signal level to ha_processed_signal level
+    # ref_signal = ref_signal * scale_factor
+
+    ref_signal = car_scene_acoustic.scale_signal_to_snr(
+        signal=ref_signal, reference_signal=processed_signal, snr=None
+    )
+
+    audio_manager.add_audios_to_save("ref_signal_normalised", ref_signal)
+
+    audio_manager.save_audios()
+
+    # 8. Compute HAAQI scores
+    aq_score_l = compute_haaqi(
+        processed_signal[0, :],
+        ref_signal[0, :],
+        audiogram_left,
+        center_frequencies,
+        sample_rate,
+    )
+    aq_score_r = compute_haaqi(
+        processed_signal[1, :],
+        ref_signal[1, :],
+        audiogram_right,
+        center_frequencies,
+        sample_rate,
+    )
+    return aq_score_l, aq_score_r
+
+
+@hydra.main(config_path="", config_name="config")
+def run_calculate_aq(config: DictConfig) -> None:
+    """Evaluate the enhanced signals using the HAAQI metric."""
+
+    # Load scenes and listeners depending on config.evaluate.split
+    scenes, listener_audiograms = load_listeners_and_scenes(config)
+
+    enhanced_folder = Path("enhanced_signals")
+    logger.info(f"Evaluating from {enhanced_folder} directory")
+
+    results_file = ResultsFile("scores.csv")
+    results_file.write_header()
+
+    # Initialize acoustic scene model
+    car_scene_acoustic = CarSceneAcoustics(
+        track_duration=30,
+        sample_rate=config.sample_rate,
+        hrtf_dir=config.path.hrtf_dir,
+        config_nalr=config.nalr,
+        config_compressor=config.compressor,
+        extend_noise=0.2,
+    )
+
+    # Iterate over scenes
+    for scene_id, current_scene in tqdm(scenes.items()):
+        # Retrieve audiograms
+        listener = current_scene["listener"]
+        listener_audiogram = listener_audiograms[listener]
+
+        # Load reference signal
+        reference_song_path = (
+            Path(config.path.music_dir)
+            / f"{current_scene['split']}"
+            / f"{current_scene['song']:06d}.mp3"
+        )
+        # Read MP3 reference signal using librosa
+        reference_signal, _ = read_mp3(
+            reference_song_path.as_posix(), sample_rate=config.sample_rate
+        )
+
+        if reference_signal.ndim == 1:
+            # If mono, duplicate to stereo
+            reference_signal = np.stack([reference_signal, reference_signal], axis=0)
+
+        # Load enhanced signal
+        enhanced_folder = Path("enhanced_signals") / config.evaluate.split
+        enhanced_song_id = f"{current_scene['listener']}_{current_scene['song']}"
+        enhanced_song_path = enhanced_folder / f"{enhanced_song_id}.wav"
+
+        # Read WAV enhanced signal using scipy.io.wavfile
+        enhanced_sample_rate, enhanced_signal = wavfile.read(enhanced_song_path)
+        enhanced_signal = enhanced_signal / 32768.0
```

Comment:
Don't know if it will ever change but could this be an option or a constant that is imported across challenges/code?

Reply:
Yes, they could.
it is basically to transform an Int16 wav file  (range [-32768, 32767]) to a float with range of [-1, 1].
Very common in Audio Signal

Reply:
Ah, a form of normalisation?

Reply:
Yeah, by convention, floating point audio data is normalised between -1 and 1.
A float audio data carries more information than int16. I think it matters the most when you consider the headphones/speakers you will use to reproduce the signal.